ui/src/app/apps/apps.service.ts

```diff
@@ -96,7 +96,8 @@ export class AppsService {
 
     return this.http.delete(AppsService.appsUrl + '/' + appRegistration.type + '/' + appRegistration.name, options)
       .map(data => {
-        this.appRegistrations.items = this.appRegistrations.items.filter(item => item.name !== appRegistration.name);
+        const index: number = this.appRegistrations.items.findIndex(item => item.name === appRegistration.name);
+        this.appRegistrations.items.splice(index, 1);
```

Comment:
Still seeing a slight bug here.  
While the items state is correct.  If I'm on page 9 of 10 and I delete an entry on page 9 which removes the last entry on page 10 the page count is still 10.  If we do a hard refresh then the page count goes to 9.